### PR TITLE
blog: Add a post introducing the new ARM tool.

### PR DESCRIPTION
**Pull request by kallisti5**

### CODE REVIEW

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  * Takes the Fedora ARM image you download
+  * You specify the target board (Like the Raspberry Pi 3)
+  * Takes the properiety bootloader artifacts from your local system
+  * Selects the correct FDT file
+  * Burns all of this to an SD card ready to boot on your device
+
+So, end users use the "arm-image-installer" tool to "burn" the ARM image to an SD card (and doing a bunch of magic under the covers)
+
+{{< alert-warning "Spoiler" "Haiku on ARM is in its infancy. Don't expect any magic. 'rpi2' will likely work the 'best' which involves loading out kernel and spitting out information to the serial uart.">}}
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Don't we have framebuffer support on the Beagle xM? That would be a little more exciting :D

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+# Rune Image
+
+With arm-image-installer in my mind, I went to work on "Rune Image". Rune is a tool written in Rust which:
+
+  * Takes a generic ["Haiku ARM" image](https://download.haiku-os.org/nightly-images/arm/) you provide (hrev51955 or later)
+  * Reads a JSON document in one of our [git repositories](https://github.com/haiku/firmware) via HTTP
+  * Grabs the needed prioperitery bits (as well as u-boot binaries + documented source locations) via HTTP
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Rust, and needing a working network connection. Couldn't it just bundle the files instead?

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+You can find a pre-compiled binary for Linux x86_64 [here](https://github.com/kallisti5/rune-image/releases/tag/v0.1.0).
+
+# Why Rust?
+
+I've had a few people ask why I wrote rune in Rust when Haiku is 99% C++. I actually have a lot of good reasons...
+
+  * Cross-platform.
+    * Rune compiles on Linux, OS X, Windows (untested), and Haiku (our Rust port needs work though).
+    * This tool is intended to be run by end users. It needs to be portable to their operating systems.
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: C++ runs on these platforms too.

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  * Low-level.
+    * Rune doesn't need mtools or other external requirements.
+    * Rune writes to FAT partitions in our images _directly_.
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: This is not Rust-specific, either. Admit you just wanted to have fun with a new language, that's totally ok :D

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  * Minimal requirements.
+    * Easy HTTP, JSON, etc, with built-in libraries.
+
+# Future
+
+Lets avoid the sensational "Haiku now runs on ARM" news.
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Is it a good idea to blog about this, then? :o)

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+Haiku has run on ARM for quite some time, these changes simply enable us to increase our development velocity while simplifying the distribution of Haiku ARM images to a single "generic ARMv7" image.
```
Comment (pulkomandy) [inline content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Yay, now I need to install Rust and compile a tool not integrated in the buildsystem just so I can test if it boots.
Do you know the most popular request (ok, that was just geist) is "give me an easy way to run the kernel in qemu in a single command, so I don't have to mess with buildsystems and obscure tools and I can play with the kernel debugger"

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md (hunk 1 above)
Comment (kallisti5) [reply content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: We do.. just need to get the FDT's in order (as well as some u-boot binaries)

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md (hunk 2 above)
Comment (kallisti5) [reply content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: No. The property bootloaders are pretty sketchy on licensing and redistribution. u-boot is also GPL-v2. *ALSO* I really don't want to maintain nightly builds for 8 different ARM boards.

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md (hunk 3 above)
Comment (kallisti5) [reply content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Sure, but you don't get "instantly portable code".

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md (hunk 4 above)
Comment (kallisti5) [reply content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Maybe.  Python couldn't write directly to FAT filesystems though :-)

File: content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md (hunk 6 above)
Comment (kallisti5) [reply content/blog/kallisti5/2018-05-19_rune_-_haiku_images_on_arm.md]: Actually, no. I provide pre-compiled self-contained binaries per the blog article. :-)
Just Linux-x86_64 for now... but soon I should auto-generated OS X and windows binaries as well.

**Comment by kallisti5** [comment]
One thing to note is I wrote this stuff pretty generically.  If someone wants to come in and write something "better" in python and C++, they can.  It's just a JSON manifest + tweaking the filesystem of the image per that manifest.

**Comment by kallisti5** [comment]
Also, these are all reasons I didn't "assume" and put rune into the Haiku github.  I wanted community feedback first :-)

**Comment by kallisti5** [comment]
*also*  in theory, you don't need rune to boot Haiku on qemu.  Rune mostly prepares the image for various proprietary boards.  qemu doesn't need u-boot.

*also* Rune will prepare an image in place... it doesn't have to write it to a block device.

*Example:*
   rune -b rpi2 haiku-arm.mmc

haiku-arm.mmc is now ready for the Raspberry Pi 2.

*Example:*
   rune -b rpi2 -i haiku-arm.mmc  haiku-arm-rpi2.mmc

haiku-arm-rpi2.mmc is now ready for the Raspberry Pi 2 and haiku-arm.mmc is unmodified.

**Pull request merged**